### DISCUSSION TITLE
shadow: Should "Tray Icon > Read-only" block the tray icon from working ever again?

### DISCUSSION THREAD

**Discussion by stdedos** [Xpra-org/xpra#4120] [Q&A]
Even when trying to use it from the computer that's being shadowed?

Also saw this on the client:
```py
2024-02-01 00:52:36,052 unknown server setting changed: readonly=True
```

**Comment by totaam** [comment]
It makes the whole session read-only, all keyboard and pointer events are rejected by the server - which does make it impossible to re-enable read-write if that's what you meant.
This is not a bug.
The tray menu is server-side for a reason: the user who owns that desktop session can prevent remote users from interfering.

**Reply by stdedos** [reply]
> Even when trying to use it from the computer that's being shadowed?

In other words: Even standing in front of the server computer, I cannot use the tray icon anymore (let alone disable the Read-Only setting)

**Reply by totaam** [reply]
No, of course you can change it back when sitting at the computer. That's the only place you can do it from.
And I've tried it repeatedly, without fail.
